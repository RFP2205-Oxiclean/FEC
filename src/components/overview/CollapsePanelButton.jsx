import React, {useState} from 'react';

const CollapsePanelButton = ({isHidden, setIsHidden}) => {

  let handleClick = function() {
    setIsHidden(!isHidden)
  }

  return <div style={{height: "2%", marginRight: "20px"}}>
    <button onClick={() => {handleClick()}} className="collapse-panel-button">COLLAPSE</button>
  </div>
}


export default CollapsePanelButton;