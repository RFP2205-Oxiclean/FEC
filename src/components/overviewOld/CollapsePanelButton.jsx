import React, { useState } from "react";

const CollapsePanelButton = ({ setIsHiding, isHiding, isHidden, setIsHidden }) => {
  let handleClick = function () {
    setIsHiding(!isHiding);
  };

  return (
    <div className="collapse-button-container">
      <button
        data-testid="collapseButton"
        onClick={() => {
          handleClick();
        }}
        className="collapse-panel-button">
        COLLAPSE
      </button>
    </div>
  );
};

export default CollapsePanelButton;
