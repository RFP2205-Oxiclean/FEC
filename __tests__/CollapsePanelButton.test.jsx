/** @jest-environment jsdom */
import React from "react";
import { createRoot } from "react-dom/client";
import { render, cleanup } from "@testing-library/react";
import { screen } from "@testing-library/dom";
import CollapsePanelButton from "../src/components/overview/CollapsePanelButton.jsx";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

const container = document.createElement("div");
const root = createRoot(container);
root.render(<div></div>);

afterEach(cleanup);
it("should test", () => {
  const { getByTestId } = render(<CollapsePanelButton isHidden={true}></CollapsePanelButton>);
  expect(getByTestId("collapseButton")).toHaveTextContent("Collapse");
});

it("matches snapshot", () => {
  const tree = renderer.create(<CollapsePanelButton isHidden={true}></CollapsePanelButton>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot 2", () => {
  const tree = renderer.create(<CollapsePanelButton isHidden={false}></CollapsePanelButton>).toJSON();
  expect(tree).toMatchSnapshot();
});
