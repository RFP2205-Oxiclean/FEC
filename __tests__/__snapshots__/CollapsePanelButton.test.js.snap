// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="collapse-button-container"
>
  <button
    className="collapse-panel-button"
    data-testid="collapseButton"
    onClick={[Function]}
  >
    COLLAPSE
  </button>
</div>
`;
